From 4440ac2856710d51229ecabea55e005864246cb1 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:37:06 -0400
Subject: [misc] mei: nfc: mei_nfc_free has to be called under lock

Message-id: <1399041470-21330-34-git-send-email-prarit@redhat.com>
Patchwork-id: 79321
O-Subject: [RHEL7.1 PATCH BZ 1088953 33/77] mei: nfc: mei_nfc_free has to be called under lock
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit 0631213f3f6ee776db2fb2136778b570d6a17cab
Author: Tomas Winkler <tomas.winkler@intel.com>
Date:   Wed Jan 8 20:57:44 2014 +0200

    mei: nfc: mei_nfc_free has to be called under lock

    nfc_nfc_free unlink clients from the device list
    and has to be called under mei mutex

    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Reviewed-by: Alexander Usyskin <alexander.usyskin@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/nfc.c b/drivers/misc/mei/nfc.c
index 0a89220..5496190 100644
--- a/drivers/misc/mei/nfc.c
+++ b/drivers/misc/mei/nfc.c
@@ -469,7 +469,9 @@ static void mei_nfc_init(struct work_struct *work)
  return;
 
 err:
+ mutex_lock(&dev->device_lock);
  mei_nfc_free(ndev);
+ mutex_unlock(&dev->device_lock);
 
  return;
 }
-- 
1.7.1